### PR TITLE
48 testing data saving functionality

### PR DESCRIPTION
**Pull request by mbunwe-victor**
Writing a unit test for the data functionality and an end-to-end test  to ensure the data scraped is accurately saved in a file.

### CODE REVIEW

**Review by stephane-segning — CHANGES_REQUESTED** [review]
I cannot approve this PR, if the related dependencies (#47) are not done

File: .gitignore
Comment (stephane-segning) [inline .gitignore:1]: This change is not good for your ticket. Please cancel it

File: error_handling.py
Comment (stephane-segning) [inline error_handling.py:1]: This change is not good for your ticket. Please cancel it

File: test_end_to_end.py
Comment (stephane-segning) [inline test_end_to_end.py:1]: This change is not good for your ticket. Please cancel it

File: validation.py
Comment (stephane-segning) [inline validation.py:1]: This change is not good for your ticket. Please cancel it

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Please cancel unrequired changes

**Comment by mbunwe-victor** [comment]
sir @stephane-segning  i have removed the changes

**Review by stephane-segning — APPROVED** [review]

**Pull request merged**
